src/evalfunc.c

```diff
@@ -4152,10 +4152,17 @@ f_expandcmd(typval_T *argvars, typval_T *rettv)
     exarg_T	eap;
     char_u	*cmdstr;
     char	*errormsg = NULL;
+    int		emsgoff = TRUE;
 
-    if (in_vim9script() && check_for_string_arg(argvars, 0) == FAIL)
+    if (in_vim9script()
+	    && (check_for_string_arg(argvars, 0) == FAIL
+		|| check_for_opt_dict_arg(argvars, 1) == FAIL))
 	return;
 
+    if (argvars[1].v_type == VAR_DICT
+	    && dict_has_key(argvars[1].vval.v_dict, "errmsg", -1))
```

Comment:
Shouldn't this be `dict_get_bool`? if `errmsg` is present but set to `v:false` then `emsgoff` should not be set to false.